### PR TITLE
fix(helm): resolve URLs and SemVers correctly

### LINKED ISSUE #1277: bug: chart dependencies requirements.yaml mentioned in developer.md do not work
helm version
v2.0.0-alpha.4

Is requirements.yaml functionality present?

Managing Dependencies with requirements.yaml does not work.  Dependent charts in this file are not loaded. 

The commands: helm dependency update and helm dep up are not present:

helm dependency update
Error: unknown command "dependency" for "helm"
Run 'helm --help' for usage.

helm dep up
Error: unknown command "dep" for "helm"

Adding dependencies through the chart subdirectory does work.

### PR DESCRIPTION
**Pull request by technosophos**
The original dependency resolution did not correctly resolve version or
URL of a dependency. Version was tracked by filename, and URL was
assumed to be absolute. This fixes both of those.

Closes #1277

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/helm/1292)

### CODE REVIEW

**Comment by jackfrancis** [comment]
LGTM pending CI

File: cmd/helm/downloader/manager.go [hunk 1]
```diff
@@ -270,21 +272,28 @@ func urlsAreEqual(a, b string) bool {
...
+							// Not a legit entry.
+							continue
 						}
-						return verEntry.URLs[0], nil
+
+						if version == "" || versionEquals(version, verEntry.Version) {
+							return normalizeURL(repourl, verEntry.URLs[0])
+						}
+
+						return normalizeURL(repourl, verEntry.URLs[0])
```
Comment (adamreese) [inline cmd/helm/downloader/manager.go]: how many levels deep are we willing to go to find a chart?

File: cmd/helm/downloader/manager_test.go [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+*/
+
+package downloader
+
+import (
+	"bytes"
+	"testing"
+
+	"k8s.io/helm/cmd/helm/helmpath"
+	//"k8s.io/helm/pkg/repo"
```
Comment (adamreese) [inline cmd/helm/downloader/manager_test.go]: ewww, commented out package

File: cmd/helm/downloader/manager.go (hunk 1 above)
Comment (technosophos) [inline cmd/helm/downloader/manager.go]: _ALL_ the way down.

I'll see if I can break that up in a related branch.

Comment (technosophos) [inline cmd/helm/downloader/manager.go]: Okay, that change is in #1293, which is directly related to this anyway.

**Pull request merged**